js/npm_import.bzl

```diff
@@ -1,6 +1,7 @@
 "repository rules for importing packages from npm"
 
 load("//js/private:translate_package_lock.bzl", lib = "translate_package_lock")
+load("@bazel_tools//tools/build_defs/repo:utils.bzl", "patch")
```

Comment:
This is what `http_archive` uses and unlike the `repository_ctx` is supports `patch_args` for git patches as well as other stuff.